Added Console App (#2373)
umbraco/UmbracoDocs: Umbraco-Heartcore/Client-Libraries/Dot-Net-Console/index.md

---
versionFrom: 8.0.0
---

# .NET Core Console Application

In this article, you can read more about the .NET Core Console Application.

We will go through the process of setting the application up, explore what you can do with the application and how you can connect to your Heartcore project on Umbraco Cloud.

:::note
In order to use this console application, you will need to have the .NET Core SDK2.2. Older or newer versions will __not work__ with the application.
:::

## Installing the Console Application

> Once you have cloned down the [Umbraco Headless Client](https://github.com/umbraco/Umbraco.Headless.Client.Net), there are two ways of running the application.

Reviewer: These "two ways of running the application", those are the next in this article?
So:
1: Using Visual Studio and
2: Using the command line

Is that correctly understood?
If it is, perhaps you could add the numbers to the headings as well? 🤔 Just an idea!

### Using Visual Studio

Open up the `.csproj` file located at `~\Umbraco.Headless.Client.Net\samples\Umbraco.Headless.Client.Samples.Console\Umbraco.Headless.Client.Samples.Console` and press F5. This will launch the Console application.

### Using the Command Line

> By using the command line, you will need to tun run the following commands from the `Umbraco.Headless.Client.Samples.Console` folder in order to run the application.

Reviewer: ```suggestion
By using the command line, you will need to run the following commands from the `Umbraco.Headless.Client.Samples.Console` folder in order to run the application.
```

```
> dotnet restore
> dotnet run
```

## Walkthrough of the Application

Once the application is running, you will need to enter the Project Alias of your Heartcore Project.

```
Booting Umbraco Headless console

Enter the Project Alias of your Headless Project

```

> In order to use this application to its fullest, you will need a Heartcore Project on Umbraco Cloud. If you do not have a project yet, you can [create a trial](https://umbraco.com/try-umbraco-heartcore/?gclid=Cj0KCQjwpLfzBRCRARIsAHuj6qXbS_nfMnUhggqPQ59ixHg61_TaRBcooqbUrbfJJ7yQOq8w7nzJphMaAmkTEALw_wcB). It is recommended that you have a project with both content and media items.

Reviewer: Removed part of the URL, as that's automatically generated by Google Analytics 😁 

```suggestion
In order to use this application to its fullest, you will need a Heartcore Project on Umbraco Cloud. If you do not have a project yet, you can [create a trial](https://umbraco.com/try-umbraco-heartcore/). It is recommended that you have a project with both content and media items.
```

If you do not have a project or trial you can also connect to `demo-headless`.

:::note
If you are connecting to a Heartcore Project with multiple environments you will have to use the alias for the __Development environment__.
:::

Once you have entered the alias of the project, you are presented with the following options

```
[A] Fetch and show Content Tree
[B] Fetch and show Media Tree
[C] Show Root Content
[D] Show Root Media
[E] List Content URLs
[F] Upload image to Media Library
[X] Exit
```

Option A - E uses the [Content Delivery API](../API-Documentation/Content-Delivery/index.md) and can be used for any Headless Project, which has public content.

> Option F uses the [Content Management API](../API-Documentation/Content-Management/index.md), so an [API Key](../Getting-Started-Cloud/Backoffice-Users-and-API-Keys/index.md) is required to run this part of the sample, as it will create a new folder in the Media Library and upload an image to a new Media item.

Reviewer: ```suggestion
Option F uses the [Content Management API](../API-Documentation/Content-Management/index.md). This means that an [API Key](../Getting-Started-Cloud/Backoffice-Users-and-API-Keys/index.md) is required to run this part of the sample, as it will create a new folder in the Media Library and upload an image to a new Media item.
```

### Examples of the fetched data

Fetching the data looks like the following.

```
Enter your choice:
a

Fetching and listing Content tree

+- Home
   +- About
```

In the example above we have fetched the Content Tree. Is shows all the Parent and Child Nodes. The Child Node is shown by an indentation.

> In the next example, we have fetched the Root media.

Reviewer: A quick comment for this sample with the Media - the _names_ of the Media items.
It's a minor thing, but in order for it to look a bit more professional it would be great to have some names here, that are relatable, like `Product Image` or `Company Logo small`.

What do you think? 🤔

---
versionFrom: 8.0.0
---

# .NET Core Console Application

In this article, you can read more about the .NET Core Console Application.

We will go through the process of setting the application up, explore what you can do with the application and how you can connect to your Heartcore project on Umbraco Cloud.

:::note
In order to use this console application, you will need to have the .NET Core SDK2.2. Older or newer versions will __not work__ with the application.
:::

## Installing the Console Application

Once you have cloned down the [Umbraco Headless Client](https://github.com/umbraco/Umbraco.Headless.Client.Net), there are two ways of running the application.

### 1. Using Visual Studio

> Open up the `.csproj` file located at `~\Umbraco.Headless.Client.Net\samples\Umbraco.Headless.Client.Samples.Console\Umbraco.Headless.Client.Samples.Console` and press F5. This will launch the Console application.

Reviewer 2: ```suggestion
Open up the `.sln` file located at `~\Umbraco.Headless.Client.Net\samples\Umbraco.Headless.Client.Samples.Console` and press F5. This will launch the Console application.
```

### 2. Using the Command Line

By using the command line, you will need to run the following commands from the `Umbraco.Headless.Client.Samples.Console` folder in order to run the application.

```
> dotnet restore
> dotnet run
```

## Walkthrough of the Application

Once the application is running, you will need to enter the Project Alias of your Heartcore Project.

```
Booting Umbraco Headless console

Enter the Project Alias of your Headless Project

```

In order to use this application to its fullest, you will need a Heartcore Project on Umbraco Cloud. If you do not have a project yet, you can [create a trial](https://umbraco.com/try-umbraco-heartcore/). It is recommended that you have a project with both content and media items.

If you do not have a project or trial you can also connect to `demo-headless`.

:::note
If you are connecting to a Heartcore Project with multiple environments you will have to use the alias for the __Development environment__.

> :::

Author: Changes made and pushed

> :::

Author: There we go 🙂
